src/circuits/etable_compact/assign.rs

```diff
@@ -97,7 +98,87 @@ impl<F: FieldExt> EventTableCommonConfig<F> {
             };
         }
 
-        // Step 2: fill Status for each eentry
+        for entry in etable.entries().iter() {
+            status_entries.push(Status {
+                eid: entry.eid,
+                moid: entry.inst.moid,
+                fid: entry.inst.fid,
+                iid: entry.inst.iid,
+                mmid: entry.inst.mmid,
+                sp: entry.sp,
+                last_jump_eid: entry.last_jump_eid,
+            });
+        }
+
+        status_entries.push(Status {
+            eid: 0,
+            moid: 0,
+            fid: 0,
+            iid: 0,
+            mmid: 0,
+            sp: 0,
+            last_jump_eid: 0,
+        });
+
+        let mut mops_in_total = 0;
+        let mut jops_in_total = 0;
+
+        for (index, entry) in etable.entries().iter().enumerate() {
+            let opcode: OpcodeClassPlain = entry.inst.opcode.clone().into();
+
+            let step_status = StepStatus {
+                current: &status_entries[index],
+                next: &status_entries[index + 1],
+            };
+
+            let config = op_configs.get(&opcode).unwrap();
+
+            config.assign(ctx, &step_status, entry)?;
+
+            mops.push(
+                config.assigned_extra_mops(ctx, &step_status, entry) + entry.inst.opcode.mops(),
+            );
+            jops.push(entry.inst.opcode.jops());
+
+            mops_in_total += mops.last().unwrap();
+            jops_in_total += jops.last().unwrap();
+
+            assign_constant!(
+                self.state,
+                EventTableCommonRangeColumnRotation::InputIndex,
+                "input index",
+                bn_to_field::<F>(&BigUint::from(host_public_inputs))
+            );
+
+            if config.is_host_public_input(&step_status, entry) {
+                host_public_inputs += 1;
+            }
+
+            for _ in 0..ETABLE_STEP_SIZE {
+                ctx.next();
+            }
+        }
+
+        ctx.reset();
+
+        mops.iter_mut().for_each(|x| {
+            let t = *x;
+            *x = mops_in_total;
+            mops_in_total -= t;
+        });
+
+        // Ignore the last return
+        jops_in_total -= 1;
+        jops.iter_mut().for_each(|x| {
+            let t = *x;
+            *x = jops_in_total;
+            jops_in_total -= t;
```

Comment:
will cause subtract from zero error. 
eg the last entry in jops is 1 and jops_in_total  is also 1
when you subtract 1 from  jops_in_total first,  jops_in_total  at line 175 will becomes zero.

Reply:
Doesn't seem to be a problem, `job_in_total` becomes to `-1` but it's a temporary variable and will not be used.

Reply:
<img width="771" alt="image" src="https://user-images.githubusercontent.com/2530711/194999454-183ec2ad-3783-4253-b71d-ece0dc1ddd9b.png">

<img width="742" alt="image" src="https://user-images.githubusercontent.com/2530711/194999409-e35203bc-4e93-4581-8a47-813d422fb9bb.png">

this test does not pass after your commit. Is  this a bug?